src/pages/workspaces/workspace/tools/WorkflowView.js

```diff
@@ -98,29 +99,56 @@ class WorkflowView extends Component {
     const { workspaceNamespace, workspaceName, workflowNamespace, workflowName } = this.props
     const workspace = Rawls.workspace(workspaceNamespace, workspaceName)
 
-    workspace.entities().then(entities =>
-      this.setState({
-        entityTypes: _.map(e => ({ value: e, label: e.replace('_', ' ') }), _.keys(entities))
-      }))
+    const entityTypes = _.map(
+      e => ({ value: e, label: e.replace('_', ' ') }),
+      _.keys(await workspace.entities())
+    )
 
     const { invalidInputs, invalidOutputs, methodConfiguration: config } =
       await workspace.methodConfig(workflowNamespace, workflowName).validate()
-    this.setState({ invalid: { inputs: invalidInputs, outputs: invalidOutputs }, config })
 
     const processIO = _.flow(
       _.update('inputs', preprocessIOList),
       _.update('outputs', preprocessIOList)
     )
 
-    this.setState({ inputsOutputs: processIO(await Rawls.methodConfigInputsOutputs(config)) })
+    const processedIO = processIO(await Rawls.methodConfigInputsOutputs(config))
+
+    this.setState({
+      inputsOutputs: processedIO,
+      ...this.createInvalidIOMap(invalidInputs, invalidOutputs, config, processedIO),
```

Comment:
nit: This function only creates a single entry, `invalid`, so the spread is slightly misleading.